tests/test_updater_top_level_update.py

```diff
@@ -434,6 +434,27 @@ def test_new_targets_expired(self) -> None:
 
         self._assert_files_exist(["root", "timestamp", "snapshot"])
 
+    def test_compute_metafile_hashes_length(self) -> None:
```

Comment:
This test is actually testing `RepositorySimulator`, something that we have not done up to now. I don't want to say a test is not needed but @jku any ideas where to place it?

Reply:
yeah should not add simulator tests in updater test files at least.

I think you could make this into an updater test that just happens to test the simulator functionality as well? I suppose running `_run_refresh()` after each update would do it right? that should fail with current code but should work with the PR, am I reading this correctly?

Reply:
@jku `_run_refresh()` is already run after all updates, but the tests don't fail because of that. 
`update_snapshot` is not enough, as it sets the hashes and length to `None` and they are not updated in `update_timestamp` after as `compute_metafile_hashes_length` is `False`, so cannot reproduce the problem with it only.
May be I should think of moving the test to let's say `TestSimulator`?

Reply:
> _run_refresh() is already run after all updates, ...

Maybe we are not talking about the same thing: I'm looking at the test you are adding and it never runs the updater at all

> but the tests don't fail because of that.

I'm assuming that if you add `_run_refresh()` calls into your test (after each update), you can make the tests fail if you run without the fix.

If that does not happen, then I need to re-evaluate this as I must have misunderstood the problem.

Reply:
oh, `update_timestamp()` alone likely does not trigger the updater failure: if you just update the timestamp then the hashes/length that are mistakenly left in snapshot_meta happen to be correct (because snapshot hashes/length have not changed).

So you have to run `update_snapshot()` so that both snapshot and timestamp get a new version (and now the snapshot_meta might contain old hashes/length if the bug is not fixed).

I was going to suggest running `python3 test_updater_top_level_update.py --dump TestRefresh.test_compute_metafile_hashes_length ` after you add some refresh calls to see the actual metadata that updater sees... but apparently this test file does not support `--dump` yet :|

Reply:
@jku with this update in the test `RepositoryError` is raised without the fix and no error is raised with it. When I remove the fix, if I assert the `RepositoryError` before refresh, then `assertIsNone` fails, which means the hashes value is not correct, as we already know.
Unfortunately the library doesn't support `assertNotRaises`, so I cannot do a clean verification and once the bug is reproduced again an error would be thrown, rather than a test failure. But it works fine this way and is kind of verification though.

Reply:
I think we can at least check if the snapshot and timestamp versions are the expected similar to the other tests. Not sure if it should be done after each refresh or only once at the end is enough.
```
self._assert_version_equals("timestamp", 3)
self._assert_version_equals("snapshot", 3)
```

Reply:
> Unfortunately the library doesn't support assertNotRaises

"assertNotRaises" is kind of on by default the default: raising inside a test is a valid way of failing the test